app/Http/Requests/BatchStoreExperienceSkill.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Models\ExperienceAward;
+use App\Models\ExperienceCommunity;
+use App\Models\ExperienceEducation;
+use App\Models\ExperiencePersonal;
+use App\Models\ExperienceSkill;
+use App\Models\ExperienceWork;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class BatchStoreExperienceSkill extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        $user = $this->user();
+        if (!$user->can('create', ExperienceSkill::class)) {
+            return false;
+        }
+
+        $experienceSkills = $this->all();
+        foreach ($experienceSkills as $experienceSkill) {
+            $experience_type = $experienceSkill['experience_type'];
+            $experience_id = (int)$experienceSkill['experience_id'];
+            $experience = null;
+            switch ($experience_type) {
+                case 'experience_work':
+                    $experience = ExperienceWork::find($experience_id);
+                    break;
+                case 'experience_award':
+                    $experience = ExperienceAward::find($experience_id);
+                    break;
+                case 'experience_community':
+                    $experience = ExperienceCommunity::find($experience_id);
+                    break;
+                case 'experience_education':
+                    $experience = ExperienceEducation::find($experience_id);
+                    break;
+                case 'experience_personal':
+                    $experience = ExperiencePersonal::find($experience_id);
+                    break;
+            }
+
+            if ($experience === null ||
+                $user === null ||
+                !$user->can('update', $experience)
+            ) {
+                return false;
+            }
+        }
```

Comment:
I think this section could be reworked as well. The ExperienceSkill -> Experience relationship is polymorphic, so we should be able to just grab all the ExperienceSkills from the database. Something like
```php
$inputIds = collect($this->all())->pluck('id')->all();
$experienceSkills = ExperienceSkill::with('experience')->whereIn('id', $inputIds)->get();

foreach ($experienceSkills as $experienceSkill) {
  if ($experience === null ||
    $user === null ||
    !$user->can('update', $experienceSkill->experience())
  ) {
    return false;
  }
}
```

Reply:
Thanks! This solution works for the batch updating so I added it into the new Form Request I made. However, for batch storing this can't work, `$experienceSkills` will always be empty since it doesn't exist in the DB yet.

Reply:
Gotcha, I noticed that on the other batch-update PR